RE-SS3D/SS3D
# Basic health, first iteration

## Summary

This PR has for objective implementation of the first iteration of the health system, following this roadmap : 
https://github.com/RE-SS3D/SS3D/issues/1192

Most is said in the road map.

To test it :

- Put your mouse cursor on a body part of your own and press F, you should take some damages and eventually a limb is removed.
- Try using "hurt" and "examinebodypart" commands to inflict different kind of damages and check they were applied correctly.
- Try using the Hit interaction by clicking on another player and pressing hit, it should inflict some damages to the head, repeat and see them die.
- See that with no hands, you can't pick up stuff.
- See that you can't run with no feet (try hurting your feet).

## Pictures/Videos (optional)



https://github.com/RE-SS3D/SS3D/assets/14344825/63f1a999-fc81-4f36-bafc-1bb6dae2a3be


## Changes to Files


## Technical Notes (optional)

Bodyparts are networked objects, but the body layers they are made of are not, so their value are server only for now (and
it might be better to leave it that way). 
They can of course be queried for specific purpose, such is done to update player speed depending on body layers damages on
foot.

## Known issues



## Fixes (optional)

## TODO

- [x] clean and organise files
- [x] comment all the code
- [x] Hide brain on clients
- [x] fix indentation
